### DISCUSSION TITLE
Trick to getting redundant boot to work with corrupted partition?

### DISCUSSION THREAD

**Discussion by arrow53** [OE4T/meta-tegra#1501] [Q&A]
I'm on an orin nano with external nvme and nanbield. I'm using `USE_REDUNDANT_FLASH_LAYOUT_DEFAULT = "1"`

If I corrupt the B partition of the kernel with `dd if=/dev/zero of=/dev/`nvme0n1p6`` and then switch ``nvbootctrl set-active-boot-slot 1` then reboot then boot fails and it drops to the UEFI shell.

Is there some sort of configuration I can use to get it to recover into the other partition?

**Comment by dwalkes** [comment]
Recovery is different than just using redundant flash layout, which I've only used for a/b software update so far.  I remember there was some logic in cboot for this related to how many failing boots needed to be attempted before recovery occurs.  I'm not up with the latest on UEFI and https://docs.nvidia.com/jetson/archives/r35.4.1/DeveloperGuide/text/SD/Bootloader/UpdateAndRedundancy.html#bootloader-implementation doesn't go into detail.  I'd start by investigating what the stock distribution does and then, if OE4T is different, try to determine why.

**Reply by arrow53** [reply]
thanks @dwalkes if I figure this out I'll report back.
